### DISCUSSION TITLE
reexecute / refetch with variables not updating

### DISCUSSION THREAD

**Discussion by TheDarkStrix** [urql-graphql/urql#1937] [Q&A] [answered]
Hi, 

I am not sure if i am understanding the documentation properly . I am basically trying to replicate "useLazyQuery" from apollo  in urql . I am initially setting pause to `true` and manually calling refetch. The issue is variables passed along with refetch is not considered .

Not sure what's wrong here. Hoping for help !

My Code : 

``` javascript
const [result, refetch] = useQuery({
    query: getUsersTable,
    variables: { limit: 0, offset: 0 },
    pause: true,
  });

  const { fetching: loading, data: usersTableData } = result;
```

Function Calling refetch : 

``` javascript
refetch({variables : { limit: pageSize, offset }});
```

Request: 
![image](https://user-images.githubusercontent.com/18639879/132553543-9c04bd86-9365-4c25-9dd4-70a208c9ce19.png)

**Comment by JoviDeCroock** [comment, marked as answer]
`variables` isn't part of the recall function signature, it only accepts an `OperationContext` https://formidable.com/open-source/urql/docs/api/urql/#usequery

An example on how to achieve what you're looking for is to leverage the reactivity pattern hooks prescribe:

```js
const [offset, setOffset] = useState(0);
const [result, refetch] = useQuery({
  query: getUsersTable,
  variables: { limit: 10, offset },
});

const { fetching: loading, data: usersTableData } = result;
const nextPage = // do something with setOffset
```

**Reply by TheDarkStrix** [reply]
hey , 
Thank you for the very quick response . This one is just an example of what i am trying to achieve . 
For instance in apollo , in case of lazy query, i could just do :

`const [functionName , {loading , data}] = useLazyQuery(query);`
and then to execute the query, would just have to do `functionName({variables: {...variables}})` 

Is there any way in urql that gives me the ability to call the function manually yet consider the parameters passed to it during the function call, i am not looking to cache the responses , i would want to be a "network-only" request every time.

**Reply by JoviDeCroock** [reply]
You can leverage the client:

```js
const client = useClient();

useEffect(async function() {
  const result = await client.query(getUsersTable, { limit: 10, offset }).toPromise();
}, [offset, client])
```

**Reply by TheDarkStrix** [reply]
I'll give that a try ! Thanks a lot
